editor/src/clj/editor/defold_project.clj

```diff
@@ -858,43 +1261,57 @@
           load-progress-span 3
           total-progress-span (+ read-progress-span load-progress-span)
           total-progress (progress/make "" total-progress-span 0)
-          deleted-node-ids (set (:delete plan))
-
-          old-resource-node-ids-by-proj-path
-          (if (coll/empty? deleted-node-ids)
-            old-nodes-by-path
-            (persistent!
-              (reduce
-                (fn [old-resource-node-ids-by-proj-path [proj-path old-node-id]]
-                  (if (contains? deleted-node-ids old-node-id)
-                    (dissoc! old-resource-node-ids-by-proj-path proj-path)
-                    old-resource-node-ids-by-proj-path))
-                (transient old-nodes-by-path)
-                old-nodes-by-path)))
+          deleted-node-id? (set (:delete plan))
+
+          old-node-id->old-node-state
+          (fn old-node-id->old-node-state [old-node-id]
+            ;; Returns the state of the supplied old-node-id as it existed in
+            ;; the graph before the resource change plan was executed.
+            (if (resource-node/loaded? old-basis old-node-id)
+              :loaded
+              :not-loaded))
+
+          old-node-id->dependency-proj-paths
+          (fn/memoize
+            (fn old-node-id->dependency-proj-paths [old-node-id]
+              (when (and (not (deleted-node-id? old-node-id))
+                         (= :loaded (old-node-id->old-node-state old-node-id)))
+                (let [resource (g/valid-node-value old-node-id :resource old-evaluation-context)]
+                  (du/measuring resource-metrics (resource/proj-path resource) :find-old-reload-dependencies
+                    (when-some [dependencies-fn (:dependencies-fn (resource/resource-type resource))]
+                      (let [save-value (g/node-value old-node-id :save-value old-evaluation-context)]
+                        (when-not (g/error? save-value)
+                          (dependencies-fn save-value)))))))))
```

Comment:
Previously `old-resource-node-dependencies`. Now returns `nil` (i.e. an empty sequence) if the old-node was deleted as part of the resource-sync, or it had not been loaded prior to the resource-sync.